cellbrowser_tools/fov_processing.py

```diff
@@ -347,23 +336,23 @@ def add_segs_to_img(self):
         #   channel_indices[3] to channel3
         pix = self.omexml.image().Pixels
         chxml = [pix.Channel(channel) for channel in self.channel_indices]
-        planes = [
-            pix.get_planes_of_channel(channel) for channel in self.channel_indices
-        ]
-        # remove channels in reverse order to preserve indices for next removal!!
-        # this assumes that channels_to_remove is in ascending order.
-        for i in reversed(channels_to_remove):
-            pix.remove_channel(i)
-        # reset all plane indices
-        for i in range(len(planes)):
-            for j in planes[i]:
-                j.set("TheC", str(i))
-        pix.set_SizeC(4)
-        chxml = [pix.Channel(channel) for channel in range(0, 4)]
-        chxml[0].set_ID("Channel:0:0")
-        chxml[1].set_ID("Channel:0:1")
-        chxml[2].set_ID("Channel:0:2")
-        chxml[3].set_ID("Channel:0:3")
+        # remove all channels and re-add them in the approved order.
```

Comment:
So am I correct in thinking the channels have an index and here you are just making sure they are in ascending order? I think you're just assigning sequential integers which would make sense, if not what are the consequences of skipping an index? (1, 2, 4, 5)

Reply:
This code is starting with a potentially 7 channel image (with dummy channels).   The channels could be in a different order, for different input images, but we have indices that tell me what data is in what channel for the 4 channels I care about.  I am making a new image with those 4 key channels always in the same order.  That is the easy part, but the trick is to bring along as much metadata as possible when I write out a new ome-tiff.

Reply:
The remove code is removing ome channels in reverse order because I think it makes it easier on the fixups that have to happen internally.  I don't know how much it really matters.